botometerAnalyser/pipelines/usersAnalysis.js

```diff
@@ -0,0 +1,225 @@
+			percentageUnknown: 100 - (sharesPercentage.percentageBot + sharesPercentage.percentageHuman),
+			percentageBot: sharesPercentage.percentageBot,
+			percentageHuman: sharesPercentage.percentageHuman,
+		},
+		users: {
+			total: uniquedCachedUsers.length,
+			percentageUnknown: 100 - (usersPercentage.percentageBot + usersPercentage.percentageHuman),
+			percentageBot: usersPercentage.percentageBot,
+			percentageHuman: usersPercentage.percentageHuman,
+		},
+		imageUrl: imageFileName
+	};
+}
+
+
+async function answerTextSearchAnalysis({ context, analysis }) {
+	const { search, requesterUsername, responseUrl } = context;
+
+	request({
+		url: responseUrl,
+		method: 'POST',
+		json: {
+			text: `@${requesterUsername} Done!`,
+			response_type: 'in_channel',
+			attachments: [
+				{
+					title: 'During the last 7 days',
+					fields: [
+						{
+							short: false,
+							title: `On the latest ${analysis.shares.total} shares of "${search}":`,
+							value: `**${analysis.shares.percentageBot}%** have a high probability to be made by bots\n**${analysis.shares.percentageHuman}%** have a high probability to be made by humans\nFor the remaining **${analysis.shares.percentageUnknown}%** it's difficult to say`
+						},
+						{
+							short: false,
+							title: `On the ${analysis.users.total} users who have written content that contains "${search}":`,
+							value: `**${analysis.users.percentageBot}%** have a high probability to be bots\n**${analysis.users.percentageHuman}%** have a high probability to be humans\nFor the remaining **${analysis.users.percentageUnknown}%** it's difficult to say`
+						},
+					],
+				},
+				{
+					title: 'Here is the distribution',
+					title_link: `${config.get('hooks.domain')}/images/botometerAnalyser/${analysis.imageUrl}.png`,
+					image_url: `${config.get('hooks.domain')}/images/botometerAnalyser/${analysis.imageUrl}.png`
+				}
+			]
+		},
+	});
+}
+
+
+async function answerRetweetAnalysis({ context, analysis }) {
+	const { screenName, tweet, responseUrl, requesterUsername } = context;
+
+	request({
+		url: responseUrl,
+		method: 'POST',
+		json: {
+			text: `@${requesterUsername} Done!`,
+			response_type: 'in_channel',
+			attachments: [
+				{
+					fields: [
+						{
+							short: false,
+							title: `On the latest ${analysis.users.total} retweets of "${tweet.text}" by @${screenName}:`,
+							value: `**${analysis.users.percentageBot}%** have a high probability to be made by bots\n**${analysis.users.percentageHuman}%** have a high probability to be made by humans\nFor the remaining **${analysis.users.percentageUnknown}%** it's difficult to say`
+						},
+					],
+				},
+				{
+					title: 'Here is the distribution',
+					title_link: `${config.get('hooks.domain')}/images/botometerAnalyser/${analysis.imageUrl}.png`,
+					image_url: `${config.get('hooks.domain')}/images/botometerAnalyser/${analysis.imageUrl}.png`
+				}
+			]
+		},
+	});
+}
+
+
+module.exports = {
+	RETWEET_ANALYSIS,
```

Comment:
If we keep this mechanism, maybe we should not export theses constants as they are only used here